src/libostree/ostree-mutable-tree.c

```diff
@@ -71,7 +84,14 @@ struct OstreeMutableTree
    * and xattrs of this directory.  This can be NULL. */
   char *metadata_checksum;
 
-  /* const char* filename -> const char* checksum */
+  /* ======== Valid for state LAZY: =========== */
+
+  /* The repo so we can look up the checksums. */
+  OstreeRepo *repo;
```

Comment:
That's going to result in a lot of refs on the repo.  🤔(In Rust we'd just have the mutable tree have a lifetime bounded by that of the repo)

But eh, it's fine for now, if we ever profile and find this is an issue we can perhaps just have the ref be maintained by the toplevel mutable tree, and children chase up to the root.

Reply:
> That's going to result in a lot of refs on the repo.

Not too many I think.  There's only one per LAZY `OstreeMutableTree`.  This means that it's only leaf directories that have refs, which initially is only 1 directory (the root).

I agree that it's a shame to churn the repo refcount, but it's unlikely to be worth fixing.